scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/ComponentImplementationInvoker.scala

```diff
@@ -34,16 +37,20 @@ trait ComponentImplementationInvoker extends Serializable {
 
 object ComponentImplementationInvoker {
 
-  val dumbImplementationInvoker: ComponentImplementationInvoker = {
-    // We return null, because this instance can be casted to Source/Sink/etc during component compilation.
-    // when invoker is used in context when we provide only definition of a component (see ModelDefinitionBuilder)
-    (_: Params, _: NodeCompilationDependencies, _: Option[ComponentImplementationSpecificInvocationContext]) => null
+  def dumbImplementationInvoker[ComponentExecutor: NotNothing: ClassTag]: ComponentImplementationInvoker = {
```

Comment:
It is not related to this change. Returning null instead of Dumb object was a kind of technical debt that hurt me during debugging of my changes.